src/Components/test/Ignitor/ElementHive.cs

```diff
@@ -0,0 +1,474 @@
+
+                return;
+            }
+
+            // First see if we have special handling for this attribute
+            if (!this.TryApplySpecialProperty(batch, elementNode, attributeName, attributeFrame))
+            {
+                // If not, treat it as a regular string-valued attribute
+                elementNode.SetAttribute(
+                  attributeName,
+                  attributeFrame.AttributeValue);
+            }
+        }
+
+        private bool TryApplySpecialProperty(RenderBatch batch, ElementNode element, string attributeName, RenderTreeFrame attributeFrame)
+        {
+            switch (attributeName)
+            {
+                case "value":
+                    return TryApplyValueProperty(element, attributeFrame);
+                case "checked":
+                    return TryApplyCheckedProperty(element, attributeFrame);
+                default:
+                    return false;
+            }
+        }
+
+
+
+        private bool TryApplyValueProperty(ElementNode element, RenderTreeFrame attributeFrame)
+        {
+            // Certain elements have built-in behaviour for their 'value' property
+            switch (element.TagName)
+            {
+                case "INPUT":
+                case "SELECT":
+                case "TEXTAREA":
+                    {
+                        var value = attributeFrame.AttributeValue;
+                        element.SetProperty("value", value);
+
+                        if (element.TagName == "SELECT")
+                        {
+                            // <select> is special, in that anything we write to .value will be lost if there
+                            // isn't yet a matching <option>. To maintain the expected behavior no matter the
+                            // element insertion/update order, preserve the desired value separately so
+                            // we can recover it when inserting any matching <option>.
+                            element.SetProperty(SelectValuePropname, value);
+                        }
+                        return true;
+                    }
+                case "OPTION":
+                    {
+                        var value = attributeFrame.AttributeValue;
+                        if (value != null)
+                        {
+                            element.SetAttribute("value", value);
+                        }
+                        else
+                        {
+                            element.RemoveAttribute("value");
+                        }
+                        return true;
+                    }
+                default:
+                    return false;
+            }
+        }
+
+        private bool TryApplyCheckedProperty(ElementNode element, RenderTreeFrame attributeFrame)
+        {
+            // Certain elements have built-in behaviour for their 'checked' property
+            if (element.TagName == "INPUT")
```

Comment:
I don't know how clean and future-proof we want to make this code (depends on our intentions around its long-term use). But if we have time, it would be interesting to see how much it can be decoupled from HTML concepts.

For example, we could have a base `ElementHive` with the majority of the logic here plus a few virtual methods. A subclass, `HtmlElementHive`, would layer on specific knowledge of HTML behaviors such as what's here.

The reason for this is that we're trying to keep the core Components library agnostic to web-specific concepts, to leave open as much flexibility as possible for it to be used with non-web rendering technologies in the future. That's not to say we're committed to one day shipping support for a non-web rendering technology, but the easier we make it for ourselves, the more capable we'd be of experimenting with it (or maybe others would).

Reply:
Talked offline, this first pass at this tool is mostly to unblock Javier/Pranav. We can revisit later.